bcl2fastq/index_summary.py

```diff
@@ -5,70 +5,117 @@
 )
 
 all_runs = index['Run'].sort_values(ascending=False).unique()
+
 all_runs = [{'label': r, 'value': r} for r in all_runs]
 
+""" Sample_run_hidden holds json split format of "Known"
+        columns: "FlowCell","Index1","Index2","LIMS IUS SWID","LaneClusterPF","LaneClusterRaw",
+                 "LaneNumber","LaneYield","QualityScoreSum","ReadNumber","Run","RunNumber","SampleID",
+                 "SampleName","SampleNumberReads","SampleYield","TrimmedBases","Yield","YieldQ30"
+                   
+
+     pruned_unknown_hidden holds json split format of "unknown"
+        columns:  "Count","LaneNumber","Index1","Index2","Run","LIMS IUS SWID
+               
+         """
+
+"""Within the main layout division there are 4 groups:
+          1. the section with the 100% bar graph
+          2.  Nested subdivision of:
+                  a) pie graph at 24%
+                  b) bar graph at 76%
+          3. Sample Run hidden
+          4. Pruned  Unknown Hidden
+            Note: span can be used for inline setting
+
+
+          Dropdown option's label set to all_runs(combination of unknown and known)
+          Dropdown multi set to false (default) - only single select possible
+                -  multiple select gives error due to inconsistent values
+          """
 layout = html.Div(children=[
     dcc.Dropdown(
         id='run_select',
         options=all_runs,
         value=all_runs[0]['value'],
+        clearable=False
     ),
+
     dcc.Graph(
         id='known_index_bar',
+
     ),
+    # Bar graph with defaults set, no change in colour etc
     html.Div([
         html.Div(
             [dcc.Graph(id='known_unknown_pie'),
              dcc.Textarea(
                  id='known_fraction',
                  style={'width': '100%'},
                  readOnly=True,
+                 # This is the textbox at the bottom, hover over to see title
                  title=(
                      "Assumptions are made about which indexes are known "
                      "or unknown. This is due to multiple bcl2fastq analyses "
                      "being used on one run. This number should be 100%."
                  ),
+
              )],
-            style={'width': '24%', 'display': 'inline-block'}
+            style={'width': '24%', 'display': 'inline-block', }
         ),
         html.Div(
             [dcc.Graph(id='unknown_index_bar')],
-            style={'width': '74%', 'display': 'inline-block', 'float': 'right'}
+            style={'width': '76%', 'display': 'inline-block', 'float': 'right'}
         ),
     ]),
+
     html.Div(id='sample_run_hidden', style={'display': 'none'}),
     html.Div(id='pruned_unknown_hidden', style={'display': 'none'}),
+
 ])
 
 try:
     from app import app
 except ModuleNotFoundError:
     import dash
+
     app = dash.Dash(__name__)
     app.layout = layout
 
 
 @app.callback(
+
     dep.Output('sample_run_hidden', 'children'),
     [dep.Input('run_select', 'value')]
+
```

Comment:
Spaces were added to this decorator, but not others. My style is not to have spaces between decorator and function.